src/embind/emval.js

```diff
@@ -12,39 +12,39 @@
 /*jslint sub:true*/ /* The symbols 'fromWireType' and 'toWireType' must be accessed via array notation to be closure-safe since craftInvokerFunction crafts functions as strings that can't be closured. */
 
 // -- jshint doesn't understand library syntax, so we need to mark the symbols exposed here
-/*global getStringOrSymbol, emval_handles, Emval, __emval_unregister, count_emval_handles, emval_symbols, __emval_decref*/
+/*global getStringOrSymbol, emval_freelist, emval_handles, emval_handles_reserved, Emval, __emval_unregister, count_emval_handles, emval_symbols, __emval_decref*/
 /*global emval_addMethodCaller, emval_methodCallers, addToLibrary, global, emval_lookupTypes, makeLegalFunctionName*/
 /*global emval_get_global*/
 
 var LibraryEmVal = {
-  $emval_handles__deps: ['$HandleAllocator'],
-  $emval_handles: "new HandleAllocator();",
+  // Stack of handles available for reuse.
+  $emval_freelist: [],
+  // Array of alternating pairs (value, refcount).
+  $emval_handles: [],
+  // Number of handles reserved for non-use (0) or common values w/o refcount.
+  $emval_handles_reserved: 5,
```

Comment:
Rather than adding this here (which will make it exist at runtime).. you do this at the top level:

```
{{{ 
globalThis.EMVAL_RESERVED_HANDLES = 5;
null;
}}}
```

And then reference the value with ```{{{ EMVAL_RESERVED_HANDLES }}}```